src/pudl_archiver/archivers/eia/eiarecs.py

```diff
@@ -0,0 +1,231 @@
+            pattern=re.compile(r"eia_457/archive/2015_(.+)\.pdf"),
+            extension="pdf",
+        ),
+    },
+    2009: {
+        "housing_characteristics": LinkSet(
+            url=_url_for(year=2009, view="characteristics"),
+            short_name="hc",
+            pattern=re.compile(r"hc(\d{1,2}\.\d{1,2})\.xlsx"),
+            extension="xlsx",
+        ),
+        "consumption & expenditures": LinkSet(
+            url=_url_for(year=2009, view="consumption"),
+            short_name="ce",
+            pattern=re.compile(r"ce(\d\.\d{1,2}[a-z]?)\.xlsx"),
+            extension="xlsx",
+        ),
+        "microdata": LinkSet(
+            url=_url_for(year=2009, view="microdata"),
+            short_name="microdata",
+            pattern=re.compile(r"csv/(.*)\.csv"),
+            extension="csv",
+        ),
+        "microdata-codebook": LinkSet(
+            url=_url_for(year=2009, view="microdata"),
+            short_name="microdata",
+            pattern=re.compile(r"(codebook.*)\.xlsx"),
+            extension="xlsx",
+        ),
+        "methodology": LinkSet(
+            url=_url_for(year=2009, view="methodology"),
+            short_name="methodology",
+            pattern=re.compile(r"/consumption/residential/reports/2015/(.+)(\.php)?"),
+            extension="html",
+            skip_if_html=False,
+        ),
+        "methodology-forms": LinkSet(
+            url="https://www.eia.gov/survey/#eia-457",
+            short_name="methodology",
+            pattern=re.compile(r"eia_457/archive/2009 (.+)\.pdf"),
+            extension="pdf",
+        ),
+    },
+}
+
+
+class EiaRECSArchiver(AbstractDatasetArchiver):
+    """EIA RECS archiver."""
+
+    name = "eiarecs"
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Download EIA-RECS resources."""
+        for year in [2020, 2015, 2009]:
+            yield self.get_year_resources(year)
+
+    def __is_html_file(self, fileobj: BytesIO) -> bool:
+        header = fileobj.read(30).lower().strip()
+        fileobj.seek(0)
+        return b"<!doctype html" in header
+
+    async def get_year_resources(self, year: int) -> list[ResourceInfo]:
+        """Download all excel tables for a year."""
+        # Loop through all download links for tables
+        tables = []
+        zip_path = self.download_directory / f"eia-recs-{year}.zip"
+        data_paths_in_archive = set()
+        # Loop through different categories of data (all .xlsx)
+        link_sets = YEAR_LINK_SETS[year]
+        for link_set in link_sets.values():
+            for table_link in await self.get_hyperlinks(link_set.url, link_set.pattern):
+                table_link = urljoin(link_set.url, table_link).strip("/")
+                logger.info(f"Fetching {table_link}")
+                match = link_set.pattern.search(table_link)
+                matched_filename = (
+                    match.group(1)
+                    .replace(".", "-")
+                    .replace(" ", "_")
```

Comment:
we've typically attempted to always use dashes instead of _ for file names
```suggestion
                    .replace(" ", "-")
                    .replace("_", "-")
```